File: source/src/main/java/org/rss/tools/mpl/global/Parameter.java
package org.rss.tools.mpl.global;

/**
 * Holds constants for Program parameters
 * 
 * @author ricardo saturnino
 *
 */
public enum Parameter {
	
	// Remember to update params.md altogether with this

	INPUT("-i", true), 
	OUTPUT("-o", true), 
	HELP("-help", false), 
	TEMP("-temp", false), 
	SHOW_BROWSER("-show", false),

	// ---------------------------------------------------

	DEBUG("-debug", false),

	TRACE("-trace", false);

	private final String text;
	
	private final boolean valueRequired;

	private Parameter(String text, boolean required) {
		this.text = text;
		this.valueRequired = required;
	}

	public static Parameter getInstance(String value) {
		for (Parameter param : values()) {
			if (param.getText().equals(value)) {
				return param;
			}
		}
		throw new UnsupportedOperationException("Parameter not supported: " + value);
	}

	public String getText() {
		return text;
	}

	public boolean isValueRequired() {
		return valueRequired;
	}
	
}

File: source/src/main/kotlin/org/rss/tools/tevim/global/Parameter.kt
package org.rss.tools.tevim.global

/**
 * Holds constants for Program parameters
 *
 * @author ricardo saturnino
 */
enum class Parameter private constructor(val text: String, val isValueRequired: Boolean) {

    // Remember to update params.md altogether with this

    INPUT("-i", true),
    OUTPUT("-o", true),
    HELP("-help", false),
    TEMP("-temp", false),
    SHOW_BROWSER("-show", false),

    // ---------------------------------------------------

    DEBUG("-debug", false),

    TRACE("-trace", false);


    companion object {

        fun getInstance(value: String): Parameter {
            for (param in values()) {
                if (param.text == value) {
                    return param
                }
            }
            throw UnsupportedOperationException("Parameter not supported: $value")
        }
    }

}
